### PR TITLE
virtualmachine: Add support for Disk Encryption Set parameter

### LINKED ISSUE #425: Ansible support enable VM Disk Encryption
##### SUMMARY
Ansible support for Azure Disk Encryption for virtual machines and virtual machine scale sets.

https://docs.microsoft.com/en-us/cli/azure/vm/encryption?view=azure-cli-latest

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
Most probably a new Ansible module

https://docs.microsoft.com/en-us/cli/azure/vm/encryption?view=azure-cli-latest
https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disk-encryption-cli-quickstart

### PR DESCRIPTION
**Pull request by lm-sig**
##### SUMMARY
I had a requirement to have virtual machines with disks (both OS and data) that used encryption. The VM module did not support provisioning or attaching any disk type with encryption enabled.

Fixes #425

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_vm_virtualmachine

##### ADDITIONAL INFORMATION
A customer may set an Azure policy that requires VMs to have disks with encryption. If you attempt to provision a virtual machine with disks without encryption your request will fail.

This change will support both encrypted or unencrypted disks. Currently it requires you to provide the full Azure resource ID for the data encryption set ID.

### CODE REVIEW

File: plugins/modules/azure_rm_virtualmachine.py [hunk 1]
```diff
@@ -203,6 +203,15 @@
             - ReadWrite
         aliases:
             - disk_caching
+    os_disk_encryption_set:
+        description:
+            - Disk encryption set for the OS disk.
+        type: dict
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualmachine.py]: Consistent with the definition of arg
```suggestion
        type: raw
```

**Review by Fred-sun — COMMENTED** [review]
line 1634, 1810 and 2066 line too long,  longer than 160 characters.

File: plugins/modules/azure_rm_virtualmachine.py (hunk 1 above)
Comment (lm-sig) [reply plugins/modules/azure_rm_virtualmachine.py]: Updated. Thanks.

**Comment by lm-sig** [comment]
I updated the commits against the latest 'dev' branch changes. Tested against Azure US Gov cloud. Still creating VMs with encrypted disks.

**Comment by xuzhang3** [comment]
@lm-sig  can you help resolve the conflict.

**Comment by lm-sig** [comment]
I am going to open a new pull request since the dev branch keeps changing.

**Pull request closed without merging**